app/controllers/campaign_pages_controller.rb

```diff
@@ -23,6 +23,14 @@ def create
     i = 0
     widgets.each do |widget_type_name, widget_data|
       widget_type_id = widget_data.delete('widget_type')
+
+
+      if defined? widget_data['checkboxes']['{cb_number}']
```

Comment:
I'm super happy about you snooping on my PRs. :grinning: 

I'm building out dynamic checkboxes/textareas by hiding them in our form layout and then using JS to copy them over. The problem here is that they still get submitted with the rest of the form data, so if you're doing a petition form, you wind up with an extra checkbox just lying around with no label and no value. Ditto a textarea.

So this line is hopping in and looking to see if the placeholder checkbox value is defined within the widget hash, and if it is, deleting it out before we save that widget's JSON.

It's not a super clean way to do it, but the alternatives aren't great either, as it would involve dropping the information for petition form checkboxes into views that have nothing to do with petition forms.